aptos-move/aptos-vm/src/verifier/event_validation.rs

```diff
@@ -119,17 +120,15 @@ pub(crate) fn validate_emit_calls(
 /// Given a module id extract all event metadata
 pub(crate) fn extract_event_metadata_from_module(
     session: &mut SessionExt,
+    resolver: &impl AptosMoveResolver,
     module_id: &ModuleId,
 ) -> VMResult<HashSet<String>> {
-    let metadata = session.load_module(module_id).map(|module| {
-        CompiledModule::deserialize_with_config(
-            &module,
-            &session.get_vm_config().deserializer_config,
-        )
-        .map(|module| aptos_framework::get_metadata_from_compiled_module(&module))
-    });
+    let metadata = session
```

Comment:
Calling `load_module` through the Move VM will result in a temporary transaction data cache to be created, preventing us from utilizing the compiled module cache in the session. 

As we discussed offline, either
1. Put this change into a separate PR and we'll continue the discussion from there
2. Or, make `Session::load_module` return CompiledModule instead of bytes so we remove the duplicated deserialization code

Reply:
Removed this change from this PR